common/web/gesture-recognizer/src/pathSegmenter.ts

```diff
@@ -0,0 +1,957 @@
+       * the F-distribution needed for validity testing.
+       */
+      get fDoF1(): number {
+        let numDoF = 3;
+        // Cases where there clearly may as well be 'no slope' at all.
+        // This saves us a degree of freedom, which is VERY useful for segmenting
+        // the boundary between a 'hold' and a 'move'.
+        if(this.host.pre.squaredSum(this.dependent) < 1e-8) {
+          numDoF--;
+        }
+        if(this.host.post.squaredSum(this.dependent) < 1e-8) {
+          numDoF--;
+        }
+
+        return numDoF;
+      }
+
+      /**
+       * Gets the degrees-of-freedom to be used for the denominator DoF parameter of
+       * the F-distribution needed for validity testing.
+       */
+      get fDoF2(): number {
+        return this.host.union.sampleCount - 2 - this.fDoF1;
+      }
+
+      /**
+       * States a statistically-founded level of confidence we may place in upholding
+       * the represented segmentation, as based on the regressions and axes under examination.
+       *
+       * If we cannot hold any confidence whatsoever (due to being unable to "reject the
+       * null hypothesis" from this specific regression), will return 0.
+       */
+      get certaintyThreshold() {
+        return 1 - FDistribution.thresholdTier(this.fStat, this.fDoF1, this.fDoF2);
+      }
+    }
+
+    constructor(pre: CumulativePathStats,
+                post: CumulativePathStats,
+                union: CumulativePathStats,
+                cumulativeEndpoint: CumulativePathStats) {
+      this.pre = pre;
+      this.post = post;
+      this.union = union;
+      this.endpoint = cumulativeEndpoint;
+    }
+
+    /**
+     * Provides a segmented-regression perspective for the represented interval and proposed
+     * split point based upon the two specified axes.
+     * @param dependent
+     * @param independent
+     * @returns
+     */
+    public segReg(dependentAxis: 'x' | 'y' | 't', independentAxis: 'x' | 'y' | 't') {
+      return new Segmentation.segmentationComparison(this, dependentAxis, independentAxis);
+    }
+
+    /**
+     * Defines our test for upholding a minimum of sub-segmentation based upon changes in the
+     * interval's coordinates over time.  This may occur even with no change in spacial direction
+     * if there is significant enough change in _speed_, as that "curves the line" when one axis
+     * is time.
+     */
+    get segmentationMerited(): boolean {
+      let totalThreshold = 0;
+      const xTest = new Segmentation.segmentationComparison(this, 'x', 't');
+      const yTest = new Segmentation.segmentationComparison(this, 'y', 't');
+
+      // const xTestConfig = this.xtFTestConfiguration;
+      // const yTestConfig = this.ytFTestConfiguration;
+
+      totalThreshold += xTest.certaintyThreshold >= 0.95 ? 2 : (xTest.certaintyThreshold >= 0.90 ? 1 : 0) ;
+      totalThreshold += yTest.certaintyThreshold >= 0.95 ? 2 : (yTest.certaintyThreshold >= 0.90 ? 1 : 0) ;
```

Comment:
Could we define constants for those two thresholds please -- in this function is fine.

Reply:
I... uh... sure?  The statistician in me's confused, though, and I've little idea what to name them that doesn't get needlessly complex.  I'll add a comment above the lines to explain where they came from, at least - they're 100% based on their underlying p-values.

```
// Our testing thresholds are for p=0.05 and p=0.10, which correspond to certainties of 95% and
// 90% that our segmentation did not arrive from random chance based on the axis being tested.
```

p-values are extraordinarily common "constants" in statistics.  They're as obvious to a statistician as `i` and `j` in loops are obvious to us.  For example, the F-distribution's table earlier in the file?  I didn't calculate those; I looked them up directly in my old stats text's reference table.  Such tables nigh-always list for at least p=0.10, p=0.05, and p=0.01, explicitly.  
- Also, for a different, yet interesting note:  if you here someone talk about the [five-nines](https://en.wikipedia.org/wiki/Five_nines) for a web service's stability, it's nigh-exactly the same underlying concept.
- Physicists use the five-nines (in p-value form) as their requirement for validating breakthroughs based on experiments.

Like, my first nomenclature choice for named constant... would be to name the constant with its value.  My second choices would be non-descript "threshold 1" and "threshold 2".  Any other names would probably confuse a statistician.

I'll hold off on actual naming of a constant for more feedback, since clarity here feels pretty tricky.

Reply:
Even constants such as `p05` and `p10` would be fine. It's not critical though. Been quite a few years since I did stats at Uni but I do remember these values.